src/main/java/com/samskivert/mustache/Mustache.java

```diff
@@ -430,25 +430,31 @@ protected static Template.Segment[] trim (Template.Segment[] segs, boolean top)
                 if (prevBlank && block.firstLeadsBlank()) {
                     if (pseg != null) segs[ii-1] = prev.trimTrailBlank();
                     block.trimFirstBlank();
+                    block._standaloneStart = true;
                 }
                 if (nextBlank && block.lastTrailsBlank()) {
                     block.trimLastBlank();
                     if (nseg != null) segs[ii+1] = next.trimLeadBlank();
+                    block._standaloneEnd = true;
                 }
             }
             // we have to indent partials if there is space before
             // they are also standalone...
             else if (seg instanceof IncludedTemplateSegment) {
+                IncludedTemplateSegment include = (IncludedTemplateSegment) seg;
                 if (prev != null && prevBlank && nextBlank) {
                     String indent = prev.indent();
+                    include._standalone = true;
                     if (! indent.equals("")) {
-                        segs[ii] = seg.indent(indent, nseg == null);
+                        include = include.indent(indent, pseg == null,nseg == null);
+                        segs[ii] = include;
                     }
+                    //segs[ii-1] = prev.trimTrailBlank();
```

Comment:
Meant to remove this comment and add a new one discussing how the previous segment should not be trimmed.